docs/pages/reference/yaml-reference.md

```diff
@@ -252,9 +254,11 @@ This documentation effectively reflects the Java object tree under [RawServiceSp
 
       * `env-key`
 
-        This may be used to customize the environment variable used to advertise this port within the task.
+        This may be used to customize the environment variable used to advertise this port within the task. This is most useful when a random dynamic port is being used, as it allows the task to know what port was allocated for it.
```

Comment:
"Customize" probably isn't the right word anymore, now that the environment variable won't be there if `env-key` is unset.